react/components/SKUSelector/components/SKUSelector.tsx

```diff
@@ -281,17 +294,15 @@ const variationNameToDisplayVariation =
       .filter(Boolean) as DisplayOption[]
 
     if (sortVariationsByLabel) {
-      const allNumbers = options.every(
-        (option: any) => !Number.isNaN(option.label)
+      const areAllNumbers = options.every(
+        (option: DisplayOption) => !Number.isNaN(Number(option.label))
       )
 
-      options.sort((a: any, b: any) => {
-        if (allNumbers) {
-          return a.label - b.label
-        }
+      const sortedOptions = areAllNumbers
+        ? sortOptionsNumerically([...options])
+        : sortOptionsAlphabetically([...options])
 
-        return a.label < b.label ? -1 : a.label > b.label ? 1 : 0
-      })
+      options = [...sortedOptions]
```

Comment:
Here the code recreate the entire array 3 times. Less performant, it is possible to maintain the previous idea of ​​sorting directly in the original array:
```
if (areAllNumbers) {
  sortOptionsNumerically(options)
} else {
  sortOptionsAlphabetically(options)
}
```

Reply:
The first approach was chosen because it adopts principles of immutability and code clarity. By using the spread operator ([...]) to create copies of the arrays before applying the sort functions, we guarantee that the original options array remains unchanged. With immutability principle we can prevent unwanted side effects and make it easier to track changes in state, although looking less performatic due to the copy of the arrays, changing directly the object might cause unexpected re-renders in the application

Reply:
Due to the lack of clarity of where it is used and the side effects, it is understandable.